sklearn/preprocessing/imputation.py

```diff
@@ -338,14 +350,13 @@ def transform(self, X):
         invalid_mask = np.isnan(statistics)
         valid_mask = np.logical_not(invalid_mask)
         valid_statistics = statistics[valid_mask]
-        valid_statistics_indexes = np.where(valid_mask)[0]
```

Comment:
Yeah, this is what `Imputation` has now. I added my two lines in `fit` to avoid recalculating this. The version I made `np.flatnonzero(valid_mask)`, as you suggest. If we revert the change to `fit`, should I stick that here?

Reply:
yes, you can use `flatnonzero` in transform.

Reply:
Ok, I'll undo the changes to `fit` in `Imputation` but replace with `flatnonzero`.

Reply:
Done!